server/service/cohort/db.js

```diff
@@ -16,16 +16,147 @@ exports.createCohort = async ({ name, user_id }) => {
+    return await withClientTransaction(async client => {
+        const result = await client.query(sql`
+            SELECT scenario.id as id
+            FROM scenario
+            INNER JOIN cohort_scenario
+                ON scenario.id = cohort_scenario.scenario_id
+            WHERE cohort_scenario.cohort_id = ${cohort_id}
+              AND scenario.deleted_at IS NULL;
+        `);
+        return result.rows.map(row => row.id);
+    });
+}
+
+async function getCohortRuns(cohort_id) {
+    return await withClientTransaction(async client => {
+        const result = await client.query(sql`
+            SELECT *
+            FROM run
+            INNER JOIN cohort_run
+                ON run.id = cohort_run.run_id
+            WHERE cohort_run.cohort_id = ${cohort_id};
+        `);
+
+        return result.rows;
+    });
+}
+
+async function getCohortUsers(cohort_id) {
+    return await withClientTransaction(async client => {
+        const result = await client.query(sql`
+            SELECT *
+            FROM users
+            INNER JOIN cohort_user_role
+                ON users.id = cohort_user_role.user_id
+            WHERE cohort_user_role.cohort_id = ${cohort_id};
+        `);
+
+        return result.rows;
+    });
+}
+
+exports.getCohort = async ({ id }) => {
+    return await withClientTransaction(async client => {
+        const result = await client.query(sql`
+            SELECT cohort.id, cohort.name, cohort.created_at, cohort_user_role.role
+            FROM cohort
+            LEFT JOIN cohort_user_role
+                ON cohort_user_role.cohort_id = cohort.id
+            WHERE cohort.id = ${id}
+        `);
+
+        const runs = await getCohortRuns(id);
+        const scenarios = await getCohortScenarios(id);
+        const users = await getCohortUsers(id);
+
+        return {
+            ...result.rows[0],
+            runs,
+            scenarios,
+            users
+        };
+    });
+};
+
+exports.getCohorts = async ({ user_id }) => {
+    return await withClientTransaction(async client => {
+        const result = await client.query(sql`
+            SELECT cohort.id, cohort.name, cohort.created_at, cohort_user_role.role
+            FROM cohort
+            LEFT JOIN cohort_user_role
+                ON cohort.id = cohort_user_role.cohort_id
+            WHERE cohort_user_role.user_id = ${user_id};
+        `);
+
+        const cohorts = [];
+        for (const row of result.rows) {
+            const runs = await getCohortRuns(row.id);
+            const scenarios = await getCohortScenarios(row.id);
+            const users = await getCohortUsers(row.id);
+            cohorts.push({
+                ...row,
+                runs,
+                scenarios,
+                users
+            });
+        }
+        return cohorts;
+    });
+};
+
+exports.setCohort = async () => {
+    // console.log('setCohort', params);
+
+    return {};
+    // return await withClientTransaction(async client => {
```

Comment:
Do we need this function if it's mostly commented out?

Reply:
This will be implemented in either the next or the following PR for Cohorts